### PR TITLE
Llfs packable ref change

### LINKED ISSUE #65: Get rid of PackableRef wrapper around payload object
This is to track changes to remove dependency on PackableRef for payload when calling Volume::append and Volume::calculated_grant_size.

### PR DESCRIPTION
**Pull request by bhaskarbora2**
Please review LLFS changes which remove usage of PackableRef from VolumeTest's ReadWriteEvents flow. This change will allow passing of payload to Volume::append and Volume::calculate_grant_size without PackableRef wrapper.    

Fix #65

### CODE REVIEW

**Review by tonyastolfi — CHANGES_REQUESTED** [review]
Mostly looks great!  I just have a couple of questions/suggestions about how we might support a higher degree of backwards compatibility, making it unnecessary to wrap args in `std::string_view`.

File: src/llfs/volume_config.test.cpp [hunk 1]
```diff
@@ -180,24 +180,24 @@ TEST_F(VolumeConfigTest, ConfigRestore)
 
     llfs::Volume& volume = **maybe_volume;
     llfs::StatusOr<batt::Grant> grant =
-        volume.reserve(volume.calculate_grant_size("alpha") +        //
-                           volume.calculate_grant_size("bravo") +    //
-                           volume.calculate_grant_size("charlie") +  //
-                           volume.calculate_grant_size("delta"),
+        volume.reserve(volume.calculate_grant_size(std::string_view("alpha")) +        //
```
[the remark this answers is not in the document]
Comment (bhaskarbora2) [reply src/llfs/volume_config.test.cpp:183]: We already have `calculate_grant_size` overload that takes `const std::string_view&`. The compatibility broke because we added a templated calculate_grant_size.

File: src/llfs/volume_config.test.cpp [hunk 2]
```diff
@@ -180,24 +180,24 @@ TEST_F(VolumeConfigTest, ConfigRestore)
...
+                           volume.calculate_grant_size(std::string_view("bravo")) +    //
+                           volume.calculate_grant_size(std::string_view("charlie")) +  //
+                           volume.calculate_grant_size(std::string_view("delta")),
                        batt::WaitForResource::kTrue);
 
     ASSERT_TRUE(grant.ok()) << BATT_INSPECT(grant.status());
 
-    alpha_slot = volume.append("alpha", *grant);
+    alpha_slot = volume.append(std::string_view("alpha"), *grant);
```
[the remark this answers is not in the document]
Comment (bhaskarbora2) [reply src/llfs/volume_config.test.cpp:191]: I was looking to see if we can hit compiler to look for the right match. We can do what you said for both these functions.

File: src/llfs/volume_config.test.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (tonyastolfi) [reply src/llfs/volume_config.test.cpp:183]: Right... I think that's due to the reason I mentioned in my analogous comment on `Volume::append`.  The compiler is probably considering the template to be a better match than the `std::string_view` overload because it can infer `T` as `const char *` and by so doing, avoid having to perform an implicit conversion by constructing a `std::string_view` temporary for the string.  In both cases, I think the solution is to use `std::enable_if_t` to specifically exclude this edge case.

[the remark this answers is not in the document]
Comment (bhaskarbora2) [reply src/llfs/volume_config.test.cpp:183]: As discussed will keep this part same and get rid of `calculate_grant_size(const std::string_view&..)`.

File: src/llfs/volume_config.test.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (bhaskarbora2) [reply src/llfs/volume_config.test.cpp:191]: Will keep this code same and remove `append(const std:::string_view&..) `overload.

**Review by tonyastolfi — APPROVED** [review]

**Review by tonyastolfi — COMMENTED** [review]
The current approach LGTM... offering comments/suggestions on how to fix the test breakage in the latest check...

File: src/llfs/volume.ipp [hunk 3]
```diff
@@ -22,6 +22,20 @@ inline StatusOr<TypedVolumeReader<T>> Volume::typed_reader(const SlotRangeSpec&
...
+template <typename T>
+u64 Volume::calculate_grant_size(const T& payload) const
+{
+  return packed_sizeof_slot(payload);
+}
+
+template <typename T>
+StatusOr<SlotRange> Volume::append(const T& payload, batt::Grant& grant)
+{
+  llfs::PackObjectAsRawData<T> packed_obj_as_raw{payload};
```
[the remark this answers is not in the document]
Comment (bhaskarbora2) [reply src/llfs/volume.ipp]: this is resolved now

**Review by tonyastolfi — APPROVED** [review]
Latest rev LGTM!
